modules/programs/starship.nix

```
@@ -136,8 +136,13 @@ in {
         }
         ${starshipCmd} init nu | save --force ${config.xdg.cacheHome}/starship/init.nu
       '';
-      extraConfig = ''
-        source ${config.xdg.cacheHome}/starship/init.nu
+      extraConfig = let
+        sourceCmd = if (versionAtLeast cfg.package.version "1.16.0") then
+          "use"
+        else
+          "source";
+      in ''
+        ${sourceCmd} ${config.xdg.cacheHome}/starship/init.nu
```

Generally speaking we avoid trying to do any package version checks. For the master branch it is always assumed that the user is using Nixpkgs unstable. So since Nixpkgs seems to have 1.16.0 I would suggest simply switching over to `use`, without any logic:

```suggestion
      extraConfig = ''
        use ${config.xdg.cacheHome}/starship/init.nu
```